azure-samples/src/main/java/com/microsoft/azure/management/compute/samples/ManageVirtualMachine.java

```diff
@@ -152,8 +152,8 @@ public static void main(String[] args) {
                 //=============================================================
                 // Update - Expand the OS drive size by 10 GB
 
-                Integer osDiskSizeInGb = vm.osDiskSize();
-                if (osDiskSizeInGb == null) {
+                int osDiskSizeInGb = vm.osDiskSize();
+                if (osDiskSizeInGb == 0) {
```

Comment:
yes this is for auto-created disks that happens when user chose to create from an image without setting disk size explicitly.